x/cash/dynamicfee.go

```diff
@@ -136,13 +136,17 @@ func (d DynamicFeeDecorator) Deliver(ctx weave.Context, store weave.KVStore, tx
 	return res, err
 }
 
-// maybeTakeFee will send fees only if they are positive, so we don't have to check IsZero() everywhere else
-func (d DynamicFeeDecorator) maybeTakeFee(store weave.KVStore, src weave.Address, amount x.Coin) error {
+// chargeFee will send fees only if they are positive. Zero fees are are a no
+// operation actions and negative fee is a missconfiguration error.
+func (d DynamicFeeDecorator) chargeFee(store weave.KVStore, src weave.Address, amount x.Coin) error {
 	if amount.IsZero() {
 		return nil
 	}
-	dest := d.getCollector(store)
-	return d.control.MoveCoins(store, src, dest, amount)
+	if !amount.IsPositive() {
```

Comment:
I am almost sure this is handled in MoveCoins, but much better to double-check than miss one

Reply:
Correct! :male_detective: